.github/workflows/e2e.yml

```diff
@@ -0,0 +1,90 @@
+name: e2e
+
+on:
+    push:
+        branches: [ master ]
+    pull_request:
+
+jobs:
+
+    e2e:
+        runs-on: ubuntu-latest
+        name: ${{ matrix.e2e }} - ${{ matrix.php }}
+        strategy:
+            fail-fast: false
+            matrix:
+                e2e:
+                    - 'e2e_004'
+                    - 'e2e_005'
+                    - 'e2e_011'
+                    - 'e2e_013'
+                    - 'e2e_014'
+                    - 'e2e_015'
+                    - 'e2e_016'
+                    - 'e2e_017'
+                    - 'e2e_018'
+                    - 'e2e_019'
+                    - 'e2e_020'
+                    - 'e2e_0211'
+                    - 'e2e_022'
+                    - 'e2e_023'
+                    - 'e2e_024'
+                    - 'e2e_025'
+                    - 'e2e_026'
+                    - 'e2e_027'
+                    - 'e2e_028'
+                    - 'e2e_029'
+                    - 'e2e_030'
+                    - 'e2e_031'
+                    - 'e2e_032'
+                php: [ '7.3', '8.0' ]
+
+                include:
+                    -   e2e: 'e2e_0210'
+                        php: '7.3'
+
+        steps:
+            -   uses: actions/checkout@v2
+                with:
+                    fetch-depth: 0
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: ${{ matrix.php }}
+                    ini-values: "phar.readonly=0"
+                    tools: composer:v2
+                    coverage: pcov
+
+            -   name: Get composer cache directory
+                id: composercache
+                run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            -   name: Cache composer dependencies
+                uses: actions/cache@v2
+                with:
+                    path: ${{ steps.composercache.outputs.dir }}
+                    key: ${{ runner.os }}-composer-cache-${{ hashFiles('**/composer.lock') }}
+                    restore-keys: ${{ runner.os }}-composer-cache-
+
+            -   name: Set composer root version
+                run: |
+                    source .composer-root-version
+                    echo "COMPOSER_ROOT_VERSION=${COMPOSER_ROOT_VERSION}" >> $GITHUB_ENV
+
+            -   name: Install dependencies
+                run: composer install --no-interaction --no-progress --no-suggest --prefer-dist
+
+            -   name: Run check-composer-root-version
+                run: make check-composer-root-version
+
+            -   name: Generate coverage
+                if: matrix.e2e == 'e2e_020'
+                run: make tc
```

Comment:
We should not rely on the order of the tests to make them pass.
I propose you another solution: https://github.com/smoench/php-scoper/pull/1/commits/c270aa95321d4d1842c0c3f71e7108e1228a0cf0

Reply:
Nice! I didn't notice you created a PR against my fork

Reply:
I cherry-picked them. Thank you!

Reply:
I just commited another fix for e2e_030

Reply:
Are you planning to fix more e2e tests? I just want to prevent work is done twice :)

Reply:
I added a task list -> https://github.com/humbug/php-scoper/pull/446#issue-529411440

Reply:
Not right now. Maybe this evening if I have time.